sdk/nodejs/appservice/zMixins.ts

```diff
@@ -847,24 +847,28 @@ declare module "./functionApp" {
     }
 }
 
-FunctionApp.prototype.getHostKeys = function(this: FunctionApp) {
+FunctionApp.prototype.getHostKeys = function(this: FunctionApp, attempts?: number) {
     return this.id.apply(async id => {
+        const retryAttempts = attempts || 5
         const credentials = await core.getServiceClientCredentials();
         const client = new AzureServiceClient(credentials);
         const url = `https://management.azure.com${id}/host/default/listkeys?api-version=2018-02-01`;
 
         const response = await client.sendRequest({ method: "POST", url });
-        if (response.status >= 400) {
-            throw new Error(`Failed to retrieve the host keys: ${response.bodyAsText}`);
+        if(response.status < 400) {
+            const body = response.parsedBody;
+            if (body.masterKey === undefined || body.systemKeys === undefined || body.functionKeys === undefined) {
+                throw new Error(`Wrong shape of the host keys response: ${response.bodyAsText}`);
+            }
+            return pulumi.output(body as FunctionHostKeys);
         }
-
-        const body = response.parsedBody;
-        if (body.masterKey === undefined || body.systemKeys === undefined || body.functionKeys === undefined) {
-            throw new Error(`Wrong shape of the host keys response: ${response.bodyAsText}`);
+        if (response.status >= 400 && retryAttempts === 0) {
+            throw new Error(`Failed to retrieve the host keys: ${response.bodyAsText}`);
         }
+        await new Promise(r => setTimeout(r, 10000));
 
-        return body as FunctionHostKeys;
-    });
+        return this.getHostKeys(retryAttempts - 1)
+    }).apply(v =>v);
```

Comment:
Consider defining a separate helper function, something like `getHostKeysWithRetries(functionAppId: string, retryAttemps: number)` that would do the recursion thing without calling `apply` on every attempt. `getHostKeys` could then become a one-liner of `apply` and a call to `getHostKeysWithRetries`.

Reply:
Separated this out into an async helper function.